src/instrumentsscene/view/staffsettingsmodel.cpp

```diff
@@ -53,27 +60,44 @@ void StaffSettingsModel::load(const QString& staffId)
     emit voicesChanged();
     emit cutawayEnabledChanged();
     emit isSmallStaffChanged();
+    emit allStaffTypesChanged();
+    emit staffTypeChanged();
 }
 
 QVariantList StaffSettingsModel::allStaffTypes() const
 {
     QVariantList result;
+    auto staff = notationParts()->staff(m_staffId);
+    if (staff == nullptr) {
+        return result;
+    }
+    int maxLines = 0;
+    bool isPercussion = false;
+    if (staff->part() && staff->part()->instrument()) {
```

Comment:
This code can be greatly simplified by adding temporary variables and reusing them instead of staff->part()->instrument()->...